cdap-unit-test/src/main/java/co/cask/cdap/test/TestBase.java

```diff
@@ -298,6 +300,8 @@ protected void configure() {
     dsOpService.startAndWait();
     datasetService = injector.getInstance(DatasetService.class);
     datasetService.startAndWait();
+    namespaceService = injector.getInstance(NamespaceService.class);
```

Comment:
Yup, I think that's the case, and that was the reason the ``BodyConsumer`` object was ``null``. I think there are other places in tests where we do not use ``AppFabricServer`` too, and I may have to add similar code there as well (saw ``DefaultStoreTest`` failing).